packages/gatsby/src/redux/reducers/nodes.js

```diff
@@ -1,7 +1,9 @@
+const removeNodesByPlugin = require(`../remove-nodes-by-plugin`)
+
 module.exports = (state = new Map(), action) => {
   switch (action.type) {
     case `DELETE_CACHE`:
-      return new Map()
+      return removeNodesByPlugin(action.payload, state, false)
```

Comment:
I think we need the same for Loki?

Reply:
Yup, and we need test suite that doesn't care about nodes backed so we can validate things against redux-nodes and loki-nodes